jmsSource/src/main/java/io/vantiq/extsrc/jmsSource/communication/JMSMessageProducer.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2019 Vantiq, Inc.
+ *
+ * All rights reserved.
+ *
+ * SPDX: MIT
+ */
+
+package io.vantiq.extsrc.jmsSource.communication;
+
+import javax.jms.BytesMessage;
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.DeliveryMode;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.MapMessage;
+import javax.jms.Message;
+import javax.jms.MessageProducer;
+import javax.jms.ObjectMessage;
+import javax.jms.Session;
+import javax.jms.StreamMessage;
+import javax.jms.TextMessage;
+import javax.naming.Context;
+import javax.naming.NamingException;
+
+public class JMSMessageProducer {
```

Comment:
is there enough commonality to build a superclass that handles `close()` and a bunch of the error handling in `setup...()`?

For JMSMessage{Listener,Producer}, should `setupWhatever()` be renamed to `open()` since that's the opposite of `close()` and it actually opens/enables/connects/whatever the `Listener`/`Producer`?